app/features/week-widget/week-widget.tsx

```diff
@@ -44,45 +45,94 @@ export function WeekWidget({
 		disabledWeek,
 	});
 
+	const { i18n } = useTranslation();
+	const [popupData, setPopupData] = useState<{
+		event: WeekEvent | null;
+		open: boolean;
+		exposures: PopupData | null;
+	}>({ event: null, open: false, exposures: null });
+
+	function togglePopup() {
+		setPopupData((p) => ({ ...p, open: !p.open }));
+	}
+
+	function navToDay() {
+		// biome-ignore lint/suspicious/noConsole: We are in development duh
+		console.log("Navigating to day");
+	}
+
+	function handleEventClick(event: WeekEvent): void {
+		const exposureData = {
+			dust: "safe",
+			noise: "safe",
+			vibration: "safe",
+		} as PopupData;
+		setPopupData({ event: event, open: true, exposures: exposureData });
+	}
+
+	const eventTitle = (event: WeekEvent) => {
+		const actualDay = event.startDate.toLocaleDateString(i18n.language);
+		const timeConfig: Intl.DateTimeFormatOptions = {
+			hour: "2-digit",
+			minute: "2-digit",
+		};
+		const start = event.startDate.toLocaleTimeString(i18n.language, timeConfig);
+		const end = event.endDate.toLocaleTimeString(i18n.language, timeConfig);
+		return `${actualDay}, from ${start} to ${end}`;
```

Comment:
This must be translated, you can add parameters to the t-function like this
```suggestion
		return t(($) => $.example, { day: actualDay, start: start, end: end });
```

And add this in the en/no file :
"example": "{{day}}, from {{start}} to {{end}}",